airflow/providers/amazon/aws/transfers/redshift_to_s3.py

```diff
@@ -141,8 +141,10 @@ def __init__(
     def _build_unload_query(
         self, credentials_block: str, select_query: str, s3_key: str, unload_options: str
     ) -> str:
+        # Un-escape already escaped queries
+        select_query = select_query.replace("''", "'")
```

Comment:
Would this break empty string literals…?

Reply:
Ouh! That's a good catch!

Reply:
I addressed it in [7d06f1e259a695fc68e16e50e9562f514dbb757d](https://github.com/apache/airflow/pull/35986/commits/7d06f1e259a695fc68e16e50e9562f514dbb757d). I also added a unit test for this use case